platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -322,6 +322,67 @@ mender_storage_get_artifact_name(char **artifact_name) {
     return ret;
 }
 
+mender_err_t
+mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
+    assert(NULL != artifact_type);
+
+    size_t artifact_type_len;
+
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, state, sizeof(state))) {
+        mender_log_error("Unable to save update state");
+        return MENDER_FAIL;
+    }
+
+    artifact_type_size = strlen(artifact_type) + 1;
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_ARTIFACT_TYPE, artifact_type, artifact_type_size)) {
+        mender_log_error("Unable to save update state");
+        return MENDER_FAIL;
+    }
+
+    return MENDER_OK;
+}
+
+mender_err_t
+mender_storage_get_update_state(mender_update_state_t *state, char **artifact_type) {
+    assert(NULL != *artifact_type);
+
+    size_t       artifact_type_size;
+    ssize_t      n_read;
+    mender_err_t ret;
+
+    n_read = nvs_read(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, NULL, 0);
+    if (0 == n_read) {
+        mender_log_debug("Update state record empty or unavailable");
+        return MENDER_NOT_FOUND;
```

Comment:
This is not correct, as the API can also return `-ENOENT` as not found. See https://github.com/mendersoftware/mender-mcu/commit/ff4d29e49bcaa0f928afb95c6e6578c809fafba1